src/range.rs

```diff
@@ -182,6 +182,72 @@ pub fn take_while1<I, F>(f: F) -> TakeWhile1<I, F>
     TakeWhile1(f, PhantomData)
 }
 
+pub struct TakeUntilRange<I>(I::Range) where I: RangeStream;
+impl<I> Parser for TakeUntilRange<I>
+    where I: RangeStream,
+          I::Range: PartialEq + ::primitives::Range
+{
+    type Input = I;
+    type Output = I::Range;
+
+    #[inline]
+    fn parse_lazy(&mut self, mut input: Self::Input) -> ConsumedResult<Self::Output, Self::Input> {
+        use ::primitives::Range;
+
+        let len = self.0.len();
+        let mut to_consume = 0;
+        let mut look_ahead_input = input.clone();
+
+        loop {
+            match look_ahead_input.clone().uncons_range(len) {
+                Ok(xs) => {
+                    if xs == self.0 {
+                        if to_consume == 0 {
+                            return EmptyOk((self.0.clone(), input));
```

Comment:
The parser shouldn't return the tag here but instead an empty range (which we can get from input.uncons_range(0))